cinema/serializers.py

```diff
@@ -1 +1,137 @@
+
+
+class GenreSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Genre
+        fields = ["id", "name"]
+
+
+class ActorSerializer(serializers.ModelSerializer):
+    full_name = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Actor
+        fields = ["id", "first_name", "last_name", "full_name"]
+
+    def get_full_name(self, obj):
+        return f"{obj.first_name} {obj.last_name}"
+
+
+class CinemaHallSerializer(serializers.ModelSerializer):
+    capacity = serializers.SerializerMethodField()
+
+    class Meta:
+        model = CinemaHall
+        fields = ["id", "name", "rows", "seats_in_row", "capacity"]
+
+    def get_capacity(self, obj):
+        # Вычисляем емкость (количество мест) в зале
+        return obj.rows * obj.seats_in_row
+
+
+class MovieListSerializer(serializers.ModelSerializer):
+    genres = serializers.SerializerMethodField()
+    actors = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Movie
+        fields = ["id", "title", "description", "duration", "genres", "actors"]
+
+    def get_genres(self, obj):
+        # Возвращаем список строк
+        return [genre.name for genre in obj.genres.all()]
+
+    def get_actors(self, obj):
+        # Возвращаем список строк с полным именем актёра
+        return [
+            f"{actor.first_name} {actor.last_name}"
+            for actor in obj.actors.all()
+        ]
+
+
+class MovieDetailSerializer(serializers.ModelSerializer):
+    genres = serializers.SerializerMethodField()
+    actors = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Movie
+        fields = ["id", "title", "description", "duration", "genres", "actors"]
+
+    def get_genres(self, obj):
+        # Возвращаем список объектов с ключом "name"
+        return [{"name": genre.name} for genre in obj.genres.all()]
+
+    def get_actors(self, obj):
+        # Возвращаем список объектов с first_name, last_name, full_name
+        return [
+            {
+                "first_name": actor.first_name,
+                "last_name": actor.last_name,
+                "full_name": f"{actor.first_name} {actor.last_name}"
+            }
+            for actor in obj.actors.all()
+        ]
+
+
+class MovieSerializer(serializers.ModelSerializer):
+    genres = serializers.PrimaryKeyRelatedField(
+        many=True, queryset=Genre.objects.all()
+    )
+    actors = serializers.PrimaryKeyRelatedField(
+        many=True, queryset=Actor.objects.all()
+    )
+
+    class Meta:
+        model = Movie
+        fields = ["id", "title", "description", "duration", "genres", "actors"]
+
+
+class MovieSessionListSerializer(serializers.ModelSerializer):
+    movie_title = serializers.CharField(source="movie.title")
+    cinema_hall_name = serializers.CharField(source="cinema_hall.name")
+    cinema_hall_capacity = serializers.IntegerField(
+        source="cinema_hall.capacity"
+    )
```

Comment:
The `cinema_hall_capacity` field is using `source="cinema_hall.capacity"`, which is incorrect. Since `capacity` is a method field in the `CinemaHallSerializer`, you should use a `SerializerMethodField` to calculate it or ensure the correct attribute is used.